doc: Add OpenOCD SWD information (#29)
TheThingsIndustries/generic-node-docs: doc/content/getting-started/programming-software/st-examples/openocd/_index.md

---
title: "OpenOCD"
description: ""
weight: 3
distributions: null
---

> [OpenOCD](http://openocd.org/) is commonly used for on-chip debugging, but can also be used to easily program executables via a CLI.

Reviewer: ```suggestion
[OpenOCD](http://openocd.org/) is commonly used for on-chip debugging, but can also be used to easily program executables via CLI.
```

<!--more-->

## Steps to reproduce

> To program a binary executable, run the following command:

Reviewer: ```suggestion
To program a binary executable, run the following command:

```

Reviewer: Where do you run this command?

Author: In the terminal, just like make

Author: If you're talking about directory location, then it doesn't matter as one of the inputs it the binary file path

Reviewer: ok, then just say "run the following command in your terminal" and that's it

```bash
openocd -f interface/stlink.cfg -f target/stm32wlx.cfg -c "program <binary-file-location> verify reset exit 0x08000000"

> ```

Reviewer: Always add a blank line before and after bash block

> Some parts of the command are optional:

Reviewer: ```suggestion
Some parts of the command above are optional:
```

- **verify** is used to verify the loaded firmware after programming;
- **reset** will start the programmed executable without having to power cycle the device;
- **exit** quits OpenOCD as it will otherwise keep running to debug the chip.

> 0x08000000 signifies the starting address of the user program. If you do not wish to input this, you can also opt to program the **elf** file (with the **.elf** filename extention) with this command:

Reviewer: ```suggestion
`0x08000000` signifies the starting address of the user program. If you do not wish to input this, you can also opt to program the **elf** file (with the **.elf** filename extension) using the following command:
```
